### PR TITLE
C# integration Docs

### PR DESCRIPTION
**Pull request by FriggaHel**
### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/visual-testing/integrations/csharp.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+:::
+
+## Introduction
+
+This guide requires an existing C# NUnit / xUnit setup.<br />
+You can alternatively take a look to our [example repository](#examples).
+
+Sauce Visual provides an library allowing integration with [WebDriver](https://www.selenium.dev/documentation/webdriver/).
+
+Sauce Visual plugin provides a library exposing a `VisualClient` object that provides actions:
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:22]: 1.
```suggestion
Sauce Visual plugin provides a library exposing a `VisualClient` class that provides actions:
```

2. should we also mention about dispose?

File: docs/visual-testing/integrations/csharp.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+- `visualClient.VisualCheck()`: Takes a screenshot and send it to Sauce Visual for comparison.
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:24]:
```suggestion
- `VisualCheck`: Takes a screenshot and sends it to Sauce Visual for comparison.
```

File: docs/visual-testing/integrations/csharp.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+- `visualClient.VisualResults()`: Waits for all diff calculations to complete and returns a summary of results.
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:25]:
```suggestion
- `VisualResults`: Waits for all diff calculations to complete and returns a summary of results.
```

File: docs/visual-testing/integrations/csharp.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  See [Test results summary](#test-results-summary) for more details about summary format and sample usage.
+
+## Quickstart
+
+### Step 1: Add Sauce Visual dependecy
+
+Add [Sauce Visual](https://www.nuget.org/packages/SauceLabs.Visual/) dependency to your `.csproj`
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:32]: normally you don't put such a code to show how to integrate - I recommend using nuget/dotnet reference instead:

```
Windows NuGet CLI:

`Install-Package SauceLabs.Visual`

Linux/Mac OS X .NET Core CLI:

`dotnet add package SauceLabs.Visual`
```

File: docs/visual-testing/integrations/csharp.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    {
+      var sauceUsername = "YOUR_USERNAME";
+      var sauceAccessKey = "YOUR_ACCESS_KEY";
+      var sauceUrl = "https://ondemand.us-west-1.saucelabs.com:443/wd/hub";
+
+      var browserOptions = new ChromeOptions();
+      browserOptions.PlatformName = "Windows 11";
+      browserOptions.BrowserVersion = "latest";
+
+      var sauceOptions = new Dictionary<string, object>();
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:78]:
```suggestion
      var sauceOptions = new Dictionary<string, string>();
```

File: docs/visual-testing/integrations/csharp.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+      sauceOptions.Add("username", sauceUsername);
+      sauceOptions.Add("accessKey", sauceAccessKey);
+      browserOptions.AddAdditionalOption("sauce:options", sauceOptions);
+
+      Driver = new RemoteWebDriver(sauceUrl, browserOptions);
+
+      VisualClient = new VisualClient(Driver, Region.UsWest1, sauceUsername, sauceAccessKey,
+          new CreateBuildOptions() { Name = "My Visual Build", Project = "csharp-project", Branch = "csharp-branch" });
+      VisualClient.CaptureDom = true;
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:87]: shouldn't we add a comment like 

// configure other options
VisualClient.CaptureDom = true

File: docs/visual-testing/integrations/csharp.md [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+    {label: 'xUnit', value: 'xUnit'},
+  ]}>
+<TabItem value="NUnit">
+
+```csharp
+    [OneTimeTearDown]
+    public void Teardown()
+    {
+        Driver?.Quit();
+        VisualClient.Cleanup().Wait();
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:139]: We cannot suggest using `Wait()`

File: docs/visual-testing/integrations/csharp.md [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+  values={[
+    {label: 'NUnit', value: 'NUnit'},
+    {label: 'xUnit', value: 'xUnit'},
+  ]}>
+<TabItem value="NUnit">
+
+```csharp
+
+    [Test]
+    public async Task SauceDemoHomePage()
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:174]:
```suggestion
    public async Task SauceDemoHomePage_ShouldWaitForUserNameLocator_TheLocatorIsAvailableOnThePage()
```

WDYT?

File: docs/visual-testing/integrations/csharp.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+
+Upon executing your tests for the first time under this step, a visual baseline is automatically created in our system. This baseline serves as the standard for all subsequent WebDriver tests. As new tests are run, they are compared to this original baseline, with any deviations highlighted to signal visual changes. These comparisons are integral for detecting any unintended visual modifications early in your development cycle. All test builds, including the initial baseline and subsequent runs, can be monitored and managed through the Sauce Labs platform at [Sauce Visual Builds](https://app.saucelabs.com/visual/builds).
+
+Remember, the baseline is established during the initial run, and any subsequent visual differences detected will be marked for review.
+
+## Advanced usage
+
+### Test results summary
+
+`VisualClient.VisualResults()` returns a summary of test results in `dictionnary<DiffStatus, int>` format where `DiffStatus` is one of the following:
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:225]:
```suggestion
`VisualClient.VisualResults()` returns a summary of test results in `Dictionnary<DiffStatus, int>` format where `DiffStatus` is one of the following:
```

File: docs/visual-testing/integrations/csharp.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+- `DiffStatus.APPROVED`: Diffs that have detected changes and have been approved
+- `DiffStatus.REJECTED`: Diffs that have detected changes and have been rejected
+
+Sample usage:
+
+```csharp
+var expectedTotalUnapprovedDiffs = 0;
+
+var results = await VisualClient.VisualResults();
+Assert.AreEqual(expectedTotalUnapprovedDiffs, results?[DiffStatus.Unapproved]);
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:239]: I think result is always defined, right?
```suggestion
Assert.AreEqual(expectedTotalUnapprovedDiffs, results[DiffStatus.Unapproved]);
```

File: docs/visual-testing/integrations/csharp.md (hunk 7 above)
Comment (FriggaHel) [reply docs/visual-testing/integrations/csharp.md:139]: That's something we can replace for NUnit, not sure about alternatives in xUnit, will dig deeper.

File: docs/visual-testing.md [hunk 11]
```diff
@@ -157,6 +157,7 @@ Sauce Visual provides integrations with different kind of frameworks:
   <h2>E2E Testing</h2>
   <ul>
       <li><a href="/visual-testing/integrations/cypress/">Cypress Integration</a></li>
+      <li><a href="/visual-testing/integrations/csharp/">C#/.Net Integration</a></li>
```
Comment (aluedeke) [inline docs/visual-testing.md:160]:
```suggestion
      <li><a href="/visual-testing/integrations/csharp/">C#/DotNet Integration</a></li>
```

File: docs/visual-testing/integrations/csharp.md [hunk 12]
```diff
@@ -0,0 +1,343 @@
+---
+sidebar_label: C#/.Net
```
Comment (aluedeke) [inline docs/visual-testing/integrations/csharp.md:2]:
```suggestion
sidebar_label: C#/DotNet
```

File: docs/visual-testing/integrations/csharp.md [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# C#/.Net WebDriver Integration
```
Comment (aluedeke) [inline docs/visual-testing/integrations/csharp.md:9]:
```suggestion
# C#/DotNet WebDriver Integration
```

File: docs/visual-testing/integrations/csharp.md [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+
+### Step 1: Add Sauce Visual dependecy
+
+Add [Sauce Visual](https://www.nuget.org/packages/SauceLabs.Visual/) dependency to your `.csproj`
+
+<Tabs
+defaultValue="windows"
+  values={[
+    {label: 'Windows NuGet CLI', value: 'windows'},
+    {label: 'Linux/macOS/.Net Core', value: 'other'},
```
Comment (aluedeke) [inline docs/visual-testing/integrations/csharp.md:38]:
```suggestion
    {label: 'Linux/macOS/DotNet Core', value: 'other'},
```

File: docs/visual-testing/integrations/csharp.md [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+    [...]
+    public async Task InitializeAsync()
+    {
+        var browserOptions = Utils.GetBrowserOptions();
+        var sauceOptions = Utils.GetSauceOptions();
+        browserOptions.AddAdditionalOption("sauce:options", sauceOptions);
+
+        var sauceUrl = Utils.GetOnDemandURL();
+        Driver = new RemoteWebDriver(sauceUrl, browserOptions);
+        Driver.ExecuteScript("sauce:job-name=xUnit C#/.Net Visual Session");
```
Comment (aluedeke) [inline docs/visual-testing/integrations/csharp.md:125]:
```suggestion
        Driver.ExecuteScript("sauce:job-name=xUnit C#/DotNet Visual Session");
```

File: docs/visual-testing/integrations/csharp.md [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+    }
+```
+
+  </TabItem>
+</Tabs>
+
+### Step 4: Configure your Sauce Labs credentials
+
+Sauce Visual relies on environment variables for authentications.<br />
+Both `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` need to be set prior starting your .Net job.
```
Comment (aluedeke) [inline docs/visual-testing/integrations/csharp.md:230]:
```suggestion
Both `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` need to be set prior starting your DotNet job.
```

**Review by aluedeke — COMMENTED** [review]
I would prefer DotNet over .net for searchability and potential better seo?

**Comment by FriggaHel** [comment]
> I would prefer DotNet over .net for searchability and potential better seo?

@aluedeke  while I do not disagree from SEO perspective, I have some concerns regarding `DotNet` written form. I never saw that form except the cli tool. Won't that make users find it weird ?

@konraddysput Do you have an opinion on this ?

**Comment by konraddysput** [comment]
> > I would prefer DotNet over .net for searchability and potential better seo?
> 
> @aluedeke while I do not disagree from SEO perspective, I have some concerns regarding `DotNet` written form. I never saw that form except the cli tool. Won't that make users find it weird ?
> 
> @konraddysput Do you have an opinion on this ?

I agree with @FriggaHel it's more natural to use .NET rather than DotNet

**Review by konraddysput — APPROVED** [review]

**Pull request merged**